lib/linter/code-path-analysis/code-path-state.js

```diff
@@ -1220,6 +1968,12 @@ class CodePathState {
         // Update state.
         const updateSegments = forkContext.makeDisconnected(-1, -1);
 
+        /*
+         * Possible bug? `continueDestSegments` is already assigned in `makeForTest`
+         * but that value is thrown away here. Both the test and the update must be
+         * executed each time through the loop so maybe `continueDestSegments` should
+         * contain both?
+         */
         context.continueDestSegments = context.updateSegments = updateSegments;
```

Comment:
I believe this is intentional. Traversal first hits `test` and sets continue segments to it for the case that there's no `update`, because in that case the execution after `continue` goes directly to `test`.

`for (init; test; ) if (body) continue;`:

![image](https://github.com/eslint/eslint/assets/44349756/c965d044-2a7d-463e-8337-aec737757a11)

If and when the traversal later hits `update`, it overwrites continue segments to it, because in that case the execution after `continue` goes to `update`, and then from `update` to `test` (but not directly from `continue` to `test`).

`for (init; test; update) if (body) continue`;

![image](https://github.com/eslint/eslint/assets/44349756/778a254f-9534-4913-8bd9-c0edf39ab018)

Reply:
Ah that makes sense! The extent to which some properties contain data and some contain state is a bit maddening.